rsocket-transport-netty/src/test/java/io/rsocket/integration/FragmentTest.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.rsocket.integration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.rsocket.AbstractRSocket;
+import io.rsocket.Payload;
+import io.rsocket.RSocket;
+import io.rsocket.RSocketFactory;
+import io.rsocket.transport.netty.client.TcpClientTransport;
+import io.rsocket.transport.netty.server.CloseableChannel;
+import io.rsocket.transport.netty.server.TcpServerTransport;
+import io.rsocket.util.DefaultPayload;
+import io.rsocket.util.RSocketProxy;
+import java.util.concurrent.ThreadLocalRandom;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class FragmentTest {
+  private static final int frameSize = 128;
+  private AbstractRSocket handler;
+  private CloseableChannel server;
+  private String message = null;
+  private String metaData = null;
+
+  @BeforeEach
+  public void startup() {
+    int randomPort = ThreadLocalRandom.current().nextInt(10_000, 20_000);
```

Comment:
This seems somewhat flaky, particularly allocating ports in the non-ephemeral range.  IDeally this would be 0 and retrieved after starting.